src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -19,13 +26,90 @@ inline server::MetadataManager* MasterMetadataServiceImpl::metadata_manager() {
   return server::MetadataManager::GetInstance();
 }
 
+inline server::ChunkServerManager& MasterMetadataServiceImpl
+    ::chunk_server_manager() {
+  return server::ChunkServerManager::GetInstance();
+}
+
 grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
     const protos::grpc::OpenFileRequest* request,
     protos::grpc::OpenFileReply* reply) {
+  // Step 1. Create file metadata
   const std::string& filename(request->filename());
   google::protobuf::util::Status status(
       metadata_manager()->CreateFileMetadata(filename));
-  return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  if (!status.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+
+  // Step 2. Create the first file chunk for this file
+  google::protobuf::util::StatusOr<std::string> chunk_or(
+      metadata_manager()->CreateChunkHandle(filename,0));
+  if (!chunk_or.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+  const std::string& chunk(chunk_or.ValueOrDie());
+
+  // Step 3. Allocate chunk servers
+  const ushort num_of_chunk_replica(3);
+  // TODO(someone): make this number configurable via config.yml. Probably not
+  // a priority right now
+  auto chunk_server_locations(
+      chunk_server_manager().AllocateChunkServer(chunk, num_of_chunk_replica));
+
+  // Prepare the InitChunkFileReply
+  *reply->mutable_metadata() = FileChunkMetadata();
+  reply->mutable_metadata()->set_chunk_handle(chunk);
+  reply->mutable_metadata()->set_version(0);
+
+  // Step 4. Coordinate with chunk servers to initialize the file chunk 
+  for(auto chunk_server_location : chunk_server_locations) {
+    const std::string server_address(chunk_server_location.server_hostname() +
```

Comment:
server address is `hostname:port`, you forgot the colon

Reply:
Good catch, thanks.